processor/filterprocessor/telemetry.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package filterprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor"
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor/internal/metadata"
+)
+
+type trigger int
+
+const (
+	triggerMetricDataPointsDropped trigger = iota
+	triggerLogsDropped
+	triggerSpansDropped
+)
+
+var (
+	typeStr                      = metadata.Type
+	processorTagKey              = tag.MustNewKey(typeStr)
+	statMetricDataPointsFiltered = stats.Int64("metrics.filtered", "Number of metric data points dropped by the filter processor", stats.UnitDimensionless)
```

Comment:
```suggestion
	statMetricDataPointsFiltered = stats.Int64("datapoints.filtered", "Number of metric data points dropped by the filter processor", stats.UnitDimensionless)
```

Reply:
Sure, and apologies for the wait, EOY crunch has been brutal.

I considered changing this name but was a bit worried that `datapoints` would be unclear, like normally you'd have logs, metrics, and traces, *OR* the actual OTTL contexts, but not both.

Do you think it'd be a better idea to add individual metrics for all OTTL contexts? i.e., span, spanevent, metric, datapoint, and log_record?

Reply:
Since we're counting the actual data dropped (whether by OTTL or by the legacy method) I think we should omit the OTTL context for now. The reason I vote for the metric to be namespaced with `datapoints` is because we are using the `DataPointCount` function to count datapoints, not metrics.

I think in the future, in a separate PR, we could add metrics for `metrics.filtered` and `span_events.filtered`.

Reply:
That's fair, done!